### PR TITLE
Deprecations

### PR DESCRIPTION
**Pull request by omanikhi**

### CODE REVIEW

**Comment by omanikhi** [comment]
How can this be tested apart from the included unit tests?

File: src/main/java/se/repos/restclient/hc/RestClientHc.java [hunk 1]
```diff
@@ -60,37 +63,40 @@ public RestClientHc(
...
-    		public ResponseHeaders handleResponse(HttpResponse response)
-    				throws ClientProtocolException, IOException {
-    			return new HcResponseHeaders(response);
-    		}
-    	});
-        httpclient.getConnectionManager().shutdown();
-        return head;
+		try (CloseableHttpClient httpclient = HttpClientBuilder.create()
+				.setDefaultCredentialsProvider(new OnDemandCredentialsProvider(auth))
+				.setRedirectStrategy(new RedirectStrategy() {
```
Comment (omanikhi) [inline src/main/java/se/repos/restclient/hc/RestClientHc.java:85]: @takesson I take it that it is the intention not to follow the redirects when it comes to the HEAD method?

**Comment by omanikhi** [comment]
Should the version be updated?

**Comment by takesson** [comment]
Looks good. 

Version should be 1.4.0-Snapshot. 

Webapp should use this if mvn install, I think.

**Review by takesson — APPROVED** [review]
Please test in your vagrant VM. 
Is there a connection pool size that can be configured? Try temporarily setting a very low number (e.g. 2) when testing to ensure that clients are released properly.

**Comment by omanikhi** [comment]
I tried using a connection pool with maximum 2 connections and then tried running a Prepare Release without any issues. Here's the code snippet of connection pooling:

`
public class RestClientHc extends RestClientUrlBase implements RestClient {

	private final PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
	private final Logger logger = LoggerFactory.getLogger(this.getClass().getName());
	
	private RestAuthentication auth;

	@Inject
	public RestClientHc(
			@Named("config:se.repos.restclient.serverRootUrl") String serverRootUrl, 
			RestAuthentication auth) {
		super(serverRootUrl);
		this.auth = auth;
		connectionManager.setMaxTotal(4);
		logger.info("RestClient configured with root url {} and authentication {}", serverRootUrl, auth);
	}
	
	@Override
	public void get(URL url, RestResponse response) throws IOException {
		try (CloseableHttpClient httpclient = HttpClientBuilder.create()
				.setDefaultCredentialsProvider(new OnDemandCredentialsProvider(auth))
				.setConnectionManager(connectionManager)
				.setConnectionManagerShared(true)
				.build()) {
`

File: src/main/java/se/repos/restclient/hc/RestClientHc.java (hunk 1 above)
Comment (takesson) [reply src/main/java/se/repos/restclient/hc/RestClientHc.java:85]: Correct. This enables using the head request to probe for redirect and get the target url.

**Pull request merged**